packages/api/src/external/carequality/shared.ts

```diff
@@ -179,3 +181,44 @@ export async function getCqAddress({
 export const cqOrgActiveSchema = z.object({
   active: z.boolean(),
 });
+
+export function cqLinkToPatientData(cqLink: CQLink): PatientData {
+  const patient = cqLink.patientResource;
+  const firstName = patient?.name.map(name => name.given).join(" ") ?? "";
+  const lastName = patient?.name.map(name => name.family).join(" ") ?? "";
+  const dob = patient?.birthDate ? buildDayjs(patient.birthDate).format(ISO_DATE) : "";
+  const genderAtBirth = cqGenderToPatientGender(patient?.gender);
+  const address =
+    patient?.address?.map(address => ({
+      zip: address.postalCode ?? "",
+      city: address.city ?? "",
+      state: address.state as USStateForAddress,
+      country: address.country ?? "",
+      addressLine1: address.line?.[0] ?? "",
+      addressLine2: address.line?.[1] ?? "",
+    })) ?? [];
+
+  const phone = patient?.telecom?.find(telecom => isPhoneValid(telecom.value ?? ""))?.value ?? "";
+  const email = patient?.telecom?.find(telecom => isEmailValid(telecom.value ?? ""))?.value ?? "";
```

Comment:
See [this](https://github.com/metriport/metriport/blob/develop/packages/core/src/external/carequality/ihe-gateway-v2/inbound/xcpd/process/xcpd-request.ts#L32-L40), we should prob use the same functions as that code (`isPhoneNumber` and `isEmail`).

I'd even go ahead and write a function on the IHE SDK so we could just call `isPhoneNumber` and `isEmail` inside those `.find()` functions - even if it's just pointing to the same one. But I'm fine w/o this now, just the previous paragraph I think its more important.